documentation/Algorithms_and_Math.md

```diff
@@ -0,0 +1,120 @@
+# Algorithms and Mathematics
+
+This document explains some of the key algorithms and mathematical principles employed in the project, particularly within the firmware.
+
+## 1. Ballistic Targeting
+
+A core feature of the fire control system is its ability to account for gravity. Simply aiming directly at a distant target will result in the water stream falling short. To hit the target, the turret must aim slightly higher. Calculating the correct angle is a classic physics problem.
```

Comment:
It also accounts for target motion, assuming the targets movement is roughly constant.